### PR TITLE
Add integration tests for AMKO federator

### PR DESCRIPTION
**Pull request by hemantavi**
Coverage:
1. With one cluster, AMKOCluster object validations:
  - Version field check
  - Cluster context field check
  - Validity of cluster list

2. With two clusters, validations:
  - Two leaders
  - Version mismatch
  - Missing AMKOCluster object in the other cluster
  - Valid federation
  - Updates to GC/GDP propagation to other cluster
  - Failure due to version mismatch or both leaders, update the
    cluster2's AMKOCluster and check if federation starts

For all the failure cases, we also check if there was any federation.

These tests use the Ginkgo framework.

Included fixes for federator:
- Check if the current cluster is part of member clusters obtained from
  `gslb-avi-config` secret object.
- Added check for no AMKOCluster objects
- Version mismatch Fix
- Removed the duplicate AMKOCluster CRD.

### CODE REVIEW

File: federator/controllers/amkocluster_controller.go [hunk 1]
```diff
@@ -239,6 +247,11 @@ func (r *AMKOClusterReconciler) ValidateMemberClusters(ctx context.Context, memb
 		if obj.Spec.IsLeader {
 			return fmt.Errorf("AMKO in cluster %s is also a leader, conflicting state", cluster.clusterName)
 		}
+
+		if obj.Spec.Version != currVersion {
+			return fmt.Errorf("version mismatch, current AMKO: %v, AMKO in cluster %s: %v", currVersion,
```
Comment (monotosh-avi) [inline federator/controllers/amkocluster_controller.go:252]: Nit: all variables are strings, so we can use %s

File: federator/controllers/test_utils.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+*/
+
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	// . "github.com/onsi/ginkgo"
```
Comment (monotosh-avi) [inline federator/controllers/test_utils.go:24]: commented out code

**Review by monotosh-avi — COMMENTED** [review]
looks good, couple of minor comments

**Review by monotosh-avi — DISMISSED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Comment by hemantavi** [comment]
build amko

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**